Ability to turn off SSL cert verification when doing requests from Wave server (e.g. Keycloak)

#### Is your feature request related to a problem? Please describe

When WAVE server is running with keycloak which uses self-signed certificates it returns the following error:
```
panic: failed connecting to OIDC provider: Get "https://192.168.4.38:8443/auth/realms/master/.well-known/openid-configuration": x509: cannot validate certificate for 192.168.4.38 because it doesn't contain any IP SANs

goroutine 1 [running]:
github.com/h2oai/wave.Run({{0x401f3d, 0x9}, {0x4013ba, 0x7}, {0x401048, 0x6}, {0x40074e, 0x1}, {0x10d2468, 0x17}, ...})
	/home/odisseas/wave/server.go:101 +0x1aa0
main.main()
	/home/odisseas/wave/cmd/wave/main.go:256 +0x166c
```

#### Describe the solution you'd like

It would be helpful if it would be possible to turn off SSL cert verification

--- a/cmd/wave/main.go
+++ b/cmd/wave/main.go
@@ -17,7 +17,6 @@ package main
 import (
 	"bufio"
 	"bytes"
-	"crypto/tls"
 	"flag"
 	"fmt"
 	"math"
@@ -76,7 +75,6 @@ func main() {
 		httpHeadersFile      string
 		createAccessKey      bool
 		listAccessKeys       bool
-		verifyTLS            bool
 		removeAccessKeyID    string
 		rawAuthScopes        string
 		rawAuthURLParams     string
@@ -99,6 +97,7 @@ func main() {
 	flag.StringVar(&conf.Compact, "compact", "", "compact AOF log")
 	stringVar(&conf.CertFile, "tls-cert-file", "", "path to certificate file (TLS only)")
 	stringVar(&conf.KeyFile, "tls-key-file", "", "path to private key file (TLS only)")
+	boolVar(&conf.SkipCertVerification, "no-tls-verify", false, "do not verify TLS certificates during external communication - DO NOT USE IN PRODUCTION")
 	stringVar(&httpHeadersFile, "http-headers-file", "", "path to a MIME-formatted file containing additional HTTP headers to add to responses from the server")
 	boolVar(&conf.Editable, "editable", false, "allow users to edit web pages")
 	stringVar(&maxRequestSize, "max-request-size", "5M", "maximum allowed size of HTTP requests to the server (e.g. 5M or 5MB or 5MiB)")
@@ -122,7 +121,6 @@ func main() {
 	stringVar(&rawAuthScopes, "oidc-scopes", "", "OIDC scopes, comma-separated (default \"openid,profile\")")
 	stringVar(&rawAuthURLParams, "oidc-auth-url-params", "", "additional URL parameters to pass during OIDC authorization, in the format \"key:value\", comma-separated, e.g. \"foo:bar,qux:42\"")
 	boolVar(&auth.SkipLogin, "oidc-skip-login", false, "do not display the login form during OIDC authorization")
-	boolVar(&verifyTLS, "verify-tls", true, "do not verify TLS certificates during external communication - DO NOT USE IN PRODUCTION")
 
 	flag.Parse()
 
@@ -252,10 +250,6 @@ func main() {
 
 	if conf.IDE {
 		conf.Proxy = true // IDE won't function without proxy
-	}
-
-	if !verifyTLS {
-		http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
 	}
 
 	wave.Run(conf)

--- a/cmd/wave/main.go
+++ b/cmd/wave/main.go
@@ -17,6 +17,7 @@ package main
 import (
 	"bufio"
 	"bytes"
+	"crypto/tls"
 	"flag"
 	"fmt"
 	"math"
@@ -75,6 +76,7 @@ func main() {
 		httpHeadersFile      string
 		createAccessKey      bool
 		listAccessKeys       bool
+		verifyTLS            bool
 		removeAccessKeyID    string
 		rawAuthScopes        string
 		rawAuthURLParams     string
@@ -120,6 +122,7 @@ func main() {
 	stringVar(&rawAuthScopes, "oidc-scopes", "", "OIDC scopes, comma-separated (default \"openid,profile\")")
 	stringVar(&rawAuthURLParams, "oidc-auth-url-params", "", "additional URL parameters to pass during OIDC authorization, in the format \"key:value\", comma-separated, e.g. \"foo:bar,qux:42\"")
 	boolVar(&auth.SkipLogin, "oidc-skip-login", false, "do not display the login form during OIDC authorization")
+	boolVar(&verifyTLS, "verify-tls", true, "do not verify TLS certificates during external communication - DO NOT USE IN PRODUCTION")
 
 	flag.Parse()
 
@@ -249,6 +252,10 @@ func main() {
 
 	if conf.IDE {
 		conf.Proxy = true // IDE won't function without proxy
+	}
+
+	if !verifyTLS {
+		http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
 	}
 
 	wave.Run(conf)

--- a/conf.go
+++ b/conf.go
@@ -48,6 +48,7 @@ type ServerConf struct {
 	Init                 string
 	Compact              string
 	CertFile             string
+	SkipCertVerification bool
 	KeyFile              string
 	Header               http.Header
 	Editable             bool
@@ -64,15 +65,15 @@ type ServerConf struct {
 }
 
 type AuthConf struct {
-	ClientID          string
-	ClientSecret      string
-	ProviderURL       string
-	RedirectURL       string
-	EndSessionURL     string
-	PostLogoutRedirectURL string	
-	Scopes            []string
-	URLParameters     [][]string
-	SkipLogin         bool
-	SessionExpiry     time.Duration
-	InactivityTimeout time.Duration
+	ClientID              string
+	ClientSecret          string
+	ProviderURL           string
+	RedirectURL           string
+	EndSessionURL         string
+	PostLogoutRedirectURL string
+	Scopes                []string
+	URLParameters         [][]string
+	SkipLogin             bool
+	SessionExpiry         time.Duration
+	InactivityTimeout     time.Duration
 }

--- a/server.go
+++ b/server.go
@@ -15,6 +15,7 @@
 package wave
 
 import (
+	"crypto/tls"
 	"encoding/json"
 	"fmt"
 	"io"
@@ -155,6 +156,9 @@ func Run(conf ServerConf) {
 		if err := http.ListenAndServe(conf.Listen, nil); err != nil {
 			echo(Log{"t": "listen_no_tls", "error": err.Error()})
 		}
+	}
+	if conf.SkipCertVerification {
+		http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
 	}
 }
 

--- a/website/docs/configuration.md
+++ b/website/docs/configuration.md
@@ -83,12 +83,12 @@ Usage of ./waved:
         path to certificate file (TLS only)
   -tls-key-file string
         path to private key file (TLS only)
+  -no-tls-verify
+        do not verify TLS certificates during external communication - DO NOT USE IN PRODUCTION
   -version
         print version and exit
   -web-dir string
         directory to serve web assets from (default "./www")
-  -verify-tls
-        do not verify TLS certificates during external communication - DO NOT USE IN PRODUCTION
 ```
 
 Supported size units (case insensitive):
@@ -140,8 +140,8 @@ H2O_WAVE_PUBLIC_DIR [2]
 H2O_WAVE_PRIVATE_DIR [2]
 H2O_WAVE_TLS_CERT_FILE
 H2O_WAVE_TLS_KEY_FILE
+H2O_WAVE_NO_TLS_VERIFY
 H2O_WAVE_WEB_DIR
-H2O_WAVE_VERIFY_TLS
 ```
 
 Notes:
@@ -155,7 +155,7 @@ Wave server serves whole directories as they are. This means that these director
 
 ### TLS verification
 
-During development, you might want to test out TLS encryption, e.g. communication between Wave server and Keycloak. The easiest thing to do is to generate a self-signed certificate. However, Wave server verifies certificates for all communication by default, thus would throw an error for a self-signed one. ***FOR DEVELOPMENT PURPOSES ONLY***, it's possible to turn off the check using either `H2O_WAVE_VERIFY_TLS` environment variable or `verify-tls` parameter.
+During development, you might want to test out TLS encryption, e.g. communication between Wave server and Keycloak. The easiest thing to do is to generate a self-signed certificate. However, Wave server verifies certificates for all communication by default, thus would throw an error for a self-signed one. ***FOR DEVELOPMENT PURPOSES ONLY***, it's possible to turn off the check using either `H2O_WAVE_NO_TLS_VERIFY` environment variable or `no-tls-verify` parameter.
 
 :::warning
 **Disabling TLS verification is a security risk.** Make sure TLS is not disabled in production environments.

--- a/website/docs/configuration.md
+++ b/website/docs/configuration.md
@@ -87,6 +87,8 @@ Usage of ./waved:
         print version and exit
   -web-dir string
         directory to serve web assets from (default "./www")
+  -verify-tls
+        do not verify TLS certificates during external communication - DO NOT USE IN PRODUCTION
 ```
 
 Supported size units (case insensitive):
@@ -139,6 +141,7 @@ H2O_WAVE_PRIVATE_DIR [2]
 H2O_WAVE_TLS_CERT_FILE
 H2O_WAVE_TLS_KEY_FILE
 H2O_WAVE_WEB_DIR
+H2O_WAVE_VERIFY_TLS
 ```
 
 Notes:
@@ -146,9 +149,17 @@ Notes:
 - [1] `1`, `t`, `true` to enable; `0`, `f`, `false` to disable (case insensitive).
 - [2] Use OS-specific path list separator to specify multiple arguments - `:` for Linux/OSX and `;` for Windows. For example, `H2O_WAVE_PUBLIC_DIR=/images/@./files/images:/downloads/@./files/downloads`.
 
-## Public/Private dirs
+### Public/Private dirs
 
 Wave server serves whole directories as they are. This means that these directories are listable by default. If you wish to turn off this behavior, simply put an empty file called `index.html` into the folder you wish to not list.
+
+### TLS verification
+
+During development, you might want to test out TLS encryption, e.g. communication between Wave server and Keycloak. The easiest thing to do is to generate a self-signed certificate. However, Wave server verifies certificates for all communication by default, thus would throw an error for a self-signed one. ***FOR DEVELOPMENT PURPOSES ONLY***, it's possible to turn off the check using either `H2O_WAVE_VERIFY_TLS` environment variable or `verify-tls` parameter.
+
+:::warning
+Make sure the TLS verification is not turned off in production environments as it's a huge security hole.
+:::
 
 ## Configuring your app
 

--- a/website/docs/configuration.md
+++ b/website/docs/configuration.md
@@ -158,7 +158,7 @@ Wave server serves whole directories as they are. This means that these director
 During development, you might want to test out TLS encryption, e.g. communication between Wave server and Keycloak. The easiest thing to do is to generate a self-signed certificate. However, Wave server verifies certificates for all communication by default, thus would throw an error for a self-signed one. ***FOR DEVELOPMENT PURPOSES ONLY***, it's possible to turn off the check using either `H2O_WAVE_VERIFY_TLS` environment variable or `verify-tls` parameter.
 
 :::warning
-Make sure the TLS verification is not turned off in production environments as it's a huge security hole.
+**Disabling TLS verification is a security risk.** Make sure TLS is not disabled in production environments.
 :::
 
 ## Configuring your app

